### PR TITLE
Gossip New Block Hashes Via Inv Message to Non-Validators

### PR DESCRIPTION
**Pull request by tholonious**

### CODE REVIEW

**Comment by tholonious** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1092?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1092** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1092?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1091** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1091?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `feature/pos-networking-and-syncing`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tholonious and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/pos_consensus.go [hunk 1]
```diff
@@ -302,13 +302,6 @@ func (fc *FastHotStuffConsensus) handleBlockProposalEvent(
 		sendMessageToRemoteNodeAsync(validator, blockProposal)
 	}
 
-	// Broadcast the block to all inbound non-validator peers. This allows them to sync
```
Comment (tholonious) [inline lib/pos_consensus.go:305]: The block propoer's steady state FastHotStuffConsensus shouldn't broadcast blocks to non-validators. Instead, all non-validator nodes should use the existing Inv message to gossip known block hashes to each other to propagate blocks across the network

File: lib/server.go [hunk 2]
```diff
@@ -2042,11 +2042,11 @@ func (srv *Server) _handleBlockAccepted(event *BlockEvent) {
 		Hash: *blockHash,
 	}
 
-	// Iterate through all the peers and relay the InvVect to them. This will only
-	// actually be relayed if it's not already in the peer's knownInventory.
-	allPeers := srv.cmgr.GetAllPeers()
-	for _, pp := range allPeers {
-		pp.AddDeSoMessage(&MsgDeSoInv{
+	// Iterate through all non-validator peers peers and relay the InvVect to them.
```
Comment (lazynina) [inline lib/server.go:2045]:
```suggestion
	// Iterate through all non-validator peers and relay the InvVect to them.
```

**Review by lazynina — APPROVED** [review]
Just a nit on a comment typo

**Pull request merged**